File: 2023/c.py
#!/usr/bin/env python3

# Problem C - Changing the Sitting Arrangement

def swap_row(A, indices):
    return [A[i] for i in indices]

def swap_col(A, indices):
    return [[row[j] for j in indices] for row in A]

def solve(n, A):
    if n % 2 == 0:
        indices = list(range(1, n, 2)) + list(range(0, n, 2))
    else:
        indices = list(range(0, n, 2)) + list(range(1, n, 2))
    B = swap_row(A, indices)
    return swap_col(B, indices)

while True:
    n = int(input())
    if n == 0:
        break
    A = [list(map(int, input().split())) for _ in range(n)]
    assert all(len(a) == n for a in A)
    B = solve(n, A)
    for row in B:
        print(' '.join(map(str, row)))

File: 2023/elixir/c.ex
# ACM ICPC 2023 domestic qualifier
# C - Changing the Sitting Arrangement

defmodule Main do
  def main do
    n = IO.gets("") |> String.trim() |> String.to_integer()
    if n != 0 do
      a = for _ <- 1..n do
            IO.gets("") |> String.split(~r{\s}, trim: true)
              |> Enum.map(&String.to_integer/1)
          end
      solve(n, a)
        |> Enum.map(&(Enum.join(&1, " ") |> IO.puts()))
      main()
    end
  end

  def solve(_n, a) do
    a |> Enum.map(&shuffle/1) |> shuffle()
  end

  # 隣の要素が距離n/2以上離れるよう並べ替える
  def shuffle(a) do
    a = List.to_tuple(a)
    n = tuple_size(a)
    Enum.concat(1..(n-1)//2, 0..(n-1)//2)
      |> Enum.map(&elem(a, &1))
  end
end

# Remove the following line when you run this code on AtCoder
Main.main
